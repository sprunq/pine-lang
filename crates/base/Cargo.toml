[package]
name = "base"
version = "0.1.0"
edition = "2021"

[dependencies]
serde = { workspace = true }
internment = { workspace = true }
codespan-reporting = { workspace = true }
