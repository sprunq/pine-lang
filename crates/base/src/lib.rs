pub mod compile_context;
pub mod file_cache;
pub mod located;
pub mod source_id;
