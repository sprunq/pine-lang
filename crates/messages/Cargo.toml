[package]
name = "messages"
version = "0.1.0"
edition = "2021"

[dependencies]
base = { path = "../base" }
serde = { workspace = true }
codespan-reporting = { workspace = true }
