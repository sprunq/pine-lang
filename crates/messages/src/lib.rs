pub mod lexer;
pub mod message;
pub mod message_stream;
pub mod parser;
