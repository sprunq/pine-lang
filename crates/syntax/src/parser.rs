use crate::{ast::Program, lexer::Lexer, token::Token};
use base::{located::Located, source_id::SourceId};
use messages::{lexer::LexerError, message::Message, parser::ParserError};

pub struct Parser {}

impl Parser {
    pub fn parse_file(source: SourceId, file: &str) -> Result<Program, Message> {
        let _lexer = Lexer::new(source, file);
        todo!()
        //     let parser = Parser::new();
        //     let parse_res = parser.parse(source, lexer);

        //     parse_res.map_err(|e| Self::uplift_parse_err(source, file, e))
    }
}
