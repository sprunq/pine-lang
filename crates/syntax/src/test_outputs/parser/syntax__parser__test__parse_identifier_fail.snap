---
source: crates/syntax/src/parser.rs
expression: parsed
---
Err(Parse(UnrecognizedToken(
  found: Located(
    source: "",
    located: Spanned(
      span: "",
      value: "1",
    ),
  ),
  expected: "identifier",
)))
